app/javascript/packages/document-capture/hooks/use-button-role.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Returns a key event handler that calls the given callback only when emitted for the desired keys.
+ *
+ * @template E
+ *
+ * @param {string[]} keys Keys to handle.
+ * @param {import('react').KeyboardEventHandler<E>} callback
+ *
+ * @return {import('react').KeyboardEventHandler<E>}
+ */
+export function filterKeyEvents(keys, callback) {
+  return function handleKeyEvent(event) {
+    if (keys.includes(event.key)) {
+      callback(event);
+    }
+  };
+}
+
+function useButtonRole() {
```

Comment:
Not really a hook in a traditional sense, but it might be something where we'd _want_ to use lifecycle behaviors. And wasn't sure a great place to otherwise put a utility like this in our current folder structure.